dask/base.py

```diff
@@ -1088,3 +1095,37 @@ def wait(x, timeout=None, return_when="ALL_COMPLETED"):
         return wait(x, timeout=timeout, return_when=return_when)
     except (ImportError, ValueError):
         return x
+
+
+def clone_key(key):
+    prefix = key_split(key)
+    if isinstance(key, str):
+        return prefix + "-" + tokenize(key)
```

Comment:
If we clone a collection twice this will produce the same keys.  We may want to use uuids (or some other random source) here instead.

Reply:
Ah, good point. I've updated things so now, by default, cloning a collection multiple times will produce collections with new keys. I also added a `seed=` keyword to `clone` to allow for deterministic cloning (which is off by default) where needed